payments-core/src/test/java/com/stripe/android/payments/bankaccount/ui/CollectBankAccountViewModelTest.kt

```diff
@@ -92,6 +98,48 @@ class CollectBankAccountViewModelTest {
         }
     }
 
+    @Test
+    fun `init - when attachToIntent is false, attach is not called for payment intent`() = runTest {
+        val viewEffect = MutableSharedFlow<CollectBankAccountViewEffect>()
+        viewEffect.test {
+            // Given
+            givenCreateAccountSessionForPaymentIntentReturns(Result.success(linkedAccountSession))
+            givenRetrieveStripeIntentReturns(Result.success(mock()))
+
+            // When
+            val viewModel = buildViewModel(viewEffect, paymentIntentConfiguration(attachToIntent = false))
+            viewModel.onConnectionsResult(
+                ConnectionsSheetResult.Completed(connectionsLinkAccountSession)
+            )
+
+            // Then
+            awaitItem()
+            awaitItem()
+            verify(attachLinkAccountSession, never()).forPaymentIntent(any(), any(), any())
+        }
+    }
+
+    @Test
+    fun `init - when attachToIntent is false, attach is not called for setup intent`() = runTest {
+        val viewEffect = MutableSharedFlow<CollectBankAccountViewEffect>()
+        viewEffect.test {
+            // Given
+            givenCreateAccountSessionForSetupIntentReturns(Result.success(linkedAccountSession))
+            givenRetrieveStripeIntentReturns(Result.success(mock()))
+
+            // When
+            val viewModel = buildViewModel(viewEffect, setupIntentConfiguration(attachToIntent = false))
+            viewModel.onConnectionsResult(
+                ConnectionsSheetResult.Completed(connectionsLinkAccountSession)
+            )
+
+            // Then
+            awaitItem()
+            awaitItem()
```

Comment:
we can `awaitComplete` maybe?

Reply:
Yea, I tried using this before, I couldn't find a way to get this to work

Reply:
awaitLastItem maybe?